### ISSUE TITLE
PoloniexBasePollingService in 4.0.0 can't work 

### ISSUE THREAD

**Issue by ezhik97** [timmolter/XChange#1236]
Some changes were affected in snapshot. I just try to recompile my project and it can't work anymore - I have an exception:
`java.lang.NoSuchFieldError: ACCEPT_EMPTY_ARRAY_AS_NULL_OBJECT
	at com.xeiam.xchange.poloniex.service.polling.PoloniexBasePollingService$1.configureObjectMapper(PoloniexBasePollingService.java:51)
	at si.mazi.rescu.serialization.jackson.JacksonMapper.<init>(JacksonMapper.java:46)
	at si.mazi.rescu.RestInvocationHandler.<init>(RestInvocationHandler.java:69)
	at si.mazi.rescu.RestProxyFactory.createProxy(RestProxyFactory.java:47)`

It seams that some snapshot modules in maven builded not from master branch but from develop one. May be version of jackson-databind, used in the xchange project, too old and does not relevant to sources anymore?

**Comment by timmolter** [comment]
We've bumped up to 4.0.1-SNAPSHOT after the 4.0.0 release. Notice the namespace change as well.

**Comment by ezhik97** [comment]
I've moved to 4.0.0 release version (and new package) - the same story.

**Comment by timmolter** [comment]
which version of rescu is on your classpath?

**Comment by ezhik97** [comment]
1.8.1

**Comment by timmolter** [comment]
OK, that's correct. 

See: <https://github.com/vkolodrevskiy/spring-social-vkontakte/issues/35>. I think your Jackson deps are outdated versions.

**Comment by ezhik97** [comment]
Yes, wafter updated jackson-databind it works. May be it may sense to add this dependency to xchange project pom?

**Comment by timmolter** [comment]
What did you have to update jackson-databind to, to get it to work? If greater than 2.5.0, we'll have rescu update its pom.

**Comment by ezhik97** [comment]
The jackson-databind is on version 2.5.4 now.

**Comment by timmolter** [comment]
@mmazi I think we might need a pom update in rescu. thoughts?

**Comment by ezhik97** [comment]
I've not discover what jar required this dependency, but I think that if we use jackson-databind functionality (in poloniex module we use DeserializationFeature class), we must declare this dependency ourself.

**Comment by timmolter** [comment]
No, look at the dependency hierarchy. Simplest is to update the rescu pom.

**Comment by mmazi** [comment]
Jackson version has been updated for rescu 1.8.2-SNAPSHOT. I suggest to upgrade the rescu version in XChange pom (currently uses 1.8.1).

**Comment by ezhik97** [comment]
@mmazi, @timmolter - great!

**Comment by mmazi** [comment]
But I thinks XChange 4.0.0 is actually incorrect. Probably 4.0.1 should be released as a hot fix with an explicit dependency on jackson 2.5.x (and possibly jackson excluded from the rescu dependency in xchange).

**Comment by timmolter** [comment]
@mmazi You'll need to push update the rescu snapshot with the change. I can do a hot fix in a few days. there could be more issues popping up. You could also consider doing a rescu release beforehand as well.

**Comment by timmolter** [comment]
@mmazi You said you updated your 1.8.2-SNAPSHOT code/jar, but I don't see any new jar or new commits.

**Comment by mmazi** [comment]
> You said you updated your 1.8.2-SNAPSHOT code/jar, but I don't see any new jar or new commits.

@timmolter It was done a while ago: https://github.com/mmazi/rescu/commit/2eec5c74e1f51db2108f84c61365155aef7cb9ff#diff-600376dffeb79835ede4a0b285078036R221

**Comment by timmolter** [comment]
OK, but I thought the jackson dependency needed to be greater than 2.5.0, like around 2.5.4 or something. rescu 1.8.1 already depends on 2.5.0 so bumping to 1.8.2-SNAPSHOT doesn't even change anything. There are many new releases: http://mvnrepository.com/artifact/com.fasterxml.jackson.core/jackson-core

**Comment by mmazi** [comment]
@timmolter 
> OK, but I thought the jackson dependency needed to be greater than 2.5.0, like around 2.5.4 or something.

No, ACCEPT_EMPTY_ARRAY_AS_NULL_OBJECT exists in Jackson 2.5.0.

> rescu 1.8.1 already depends on 2.5.0 so bumping to 1.8.2-SNAPSHOT doesn't even change anything. 

You're right; I was wrong before.

So now I think XChange 4.0.0 is actually OK. I've recompiled (with the rescu dependency downloaded by mvn from Maven Central) with no problems.

@ezhik97 
> I've not discover what jar required this dependency,

You can check where it comes from like this:

    mvn dependency:tree 

> I think that if we use jackson-databind functionality (in poloniex module we use DeserializationFeature class), we must declare this dependency ourself.

Exactly my thoughts. @timmolter  XChange should probably have an explicit dependency on Jackson (as opposed to an implicit/transitive dependency as it stands now), because Jackson is explicitly used in XChange. 

> @mmazi I think we might need a pom update in rescu.

I don't think it's necessary now, because rescu doesn't use the new jackson features. But XChange does so it should define the jackson dependency explicitly. (This could potentially bring some compatibility problems in future, but I'll update the Jackson version in rescu with next release to prevent this.)

tl;dr: Everything works now as it is, but XChange should probably still declare an explicit Jackson dependency: at least 2.5.0, which is the safe choice, but probably the highest that doesn't cause conflicts with rescu's Jackson version.
